src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/dao/HttpGoogleBillingDAO.scala

```diff
@@ -0,0 +1,337 @@
+     */
+    Future.successful(project)
+  }
+
+  def scrubBillingProject(projectName: String): Future[Unit] = {
+    //start these early so they're async
+    val cleanupPolicyF = retryWhen500orGoogleError(() => {
+      val policyRequest = new SetIamPolicyRequest().setPolicy(new Policy().setBindings(null))
+      googleRq(cloudResources.projects().setIamPolicy(projectName, policyRequest))
+    })
+    val cleanupSAKeysF = cleanupPetSAKeys(projectName)
+
+    for {
+      project <- getGoogleProject(projectName)
+      _ <- removePermissionsFromCromwellAuthBucket(projectName, project.getProjectNumber)
+      _ <- cleanupPolicyF
+      _ <- cleanupSAKeysF
+    } yield {
+      //nah
+    }
+  }
+
+  //poll google for what's going on
+  def pollOperation(operation: ActiveOperationRecord): Future[ActiveOperationRecord] = {
+
+    // this code is a colossal DRY violation but because the operations collection is different
+    // for cloudResManager and servicesManager and they return different but identical Status objects
+    // there is not much else to be done... too bad scala does not have duck typing.
+    BillingProjectStatus.withNameIgnoreCase(operation.operationType) match {
+      case CreatingProject =>
+        retryWhen500orGoogleError(() => {
+          executeGoogleRequest(cloudResources.operations().get(operation.operationId))
+        }).map { op =>
+          operation.copy(done = toScalaBool(op.getDone), errorMessage = Option(op.getError).map(error => toErrorMessage(error.getMessage, error.getCode)))
+        }
+
+      case EnablingServices =>
+        retryWhen500orGoogleError(() => {
+          executeGoogleRequest(servicesManager.operations().get(operation.operationId))
+        }).map { op =>
+          operation.copy(done = toScalaBool(op.getDone), errorMessage = Option(op.getError).map(error => toErrorMessage(error.getMessage, error.getCode)))
+        }
+    }
+  }
+
+  case class GoogleProjectConflict(projectName: String)
+    extends GPAllocException(s"A google project by the name $projectName already exists", StatusCodes.Conflict)
+
+  //part 1
+  def createProject(projectName: String, billingAccount: String): Future[ActiveOperationRecord] = {
+    retryWhen500orGoogleError(() => {
+      executeGoogleRequest(cloudResources.projects().create(
+        new Project()
+          .setName(projectName)
+          .setProjectId(projectName)
+          .setLabels(Map("billingaccount" -> billingAccount).asJava)))
+    }).recover {
+      case t: HttpResponseException if StatusCode.int2StatusCode(t.getStatusCode) == StatusCodes.Conflict =>
+        throw GoogleProjectConflict(projectName)
+    } map ( googleOperation => {
+      if (toScalaBool(googleOperation.getDone) && Option(googleOperation.getError).exists(_.getCode == Code.ALREADY_EXISTS.value())) {
+        throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"A google project by the name $projectName already exists"))
```

Comment:
wouldn't we just throw a GoogleProjectConflict here as well?

Reply:
yeah i think so. this file needs a lot of love